doc/neomake.txt

```diff
@@ -258,6 +266,43 @@ section of the vim-airline |statusline|. Defaults to 1.
 This list is non-exhaustive at the moment, but you may find some of these
 functions useful.
 
+*neomake#Make(filemode,* *makers[,* *callback])*
+This function is called by |:Neomake(!)| command. It runs all the *makers*
+specified, in order. If *filemode* is 1, then the current file is used as input
+to the makers. |neomake#Make| returns an array of the job ids that it creates
+via 'jobstart'; you can potentially cancel these jobs with |neomake#CancelJob|.
+
+It also accepts a third, optional *callback* argument that
+is called when the command exists.  The *callback* is given the following
+dictionary as its sole argument: >
+    { 'status': <exit status of maker>,
+    \ 'name': <maker name>,
+    \ 'has_next': <true if another maker follows, false otherwise> }
+
+*neomake#Sh(command[,* *callback])*
+This function is called by the |:NeomakeSh| command. It runs the specified
+shell *command* according to 'shell'. |neomake#Sh| returns the single job id
+that it creates via 'jobstart'; you can potentially cancel this job with
+|neomake#CancelJob|.
+
+It also accepts a second, optional
+*callback* argument that is called when the command exists. The *callback* is
+given the following dictionary as its sole argument: >
+    { 'status': <exit status of command>,
+    \ 'name': 'sh: <command>',
```

Comment:
Instead of adding a "sh: " prefix to the name, why not make another property on the dict like: `'sh': <true if called from :Sh>`?